### PR TITLE
util: Reduce memory barriers and integer divisions in the work queue.

### PR DESCRIPTION
**Pull request by pcwalton**
Improves scaling on multicore ARM devices.

r? @glennw

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/6345)

### CODE REVIEW

**Comment by glennw** [comment]
Looks good to me - @michaelwu ?

**Comment by michaelwu** [comment]
The changes make sense to me, though I don't know the work stealing algorithm in much detail.

**Comment by pcwalton** [comment]
"LayoutWorker worker 3/6' panicked at 'could not initialize thread_rng: Too many open files (os error 24)'". What.

**Comment by pcwalton** [comment]
@larsbergstrom Help? Seems bots are running out of file descriptors.

**Comment by larsbergstrom** [comment]
@pcwalton Thanks for tracking that down - it appears our upgrade to OSX 10.10.3 has caused this to happen and we will either have to reduce the number of parallel threads for the WPT tests or increase the ulimt for open file handles.

**Comment by larsbergstrom** [comment]
Probably going to increase the limits machine-wide:
http://superuser.com/questions/302754/increase-the-maximum-number-of-open-file-descriptors-in-snow-leopard

CC: @edunham

**Comment by larsbergstrom** [comment]
Yeah, the default is not gonna work for us:
```
administrators-Mac-mini:~ administrator$ ulimit -a
...
open files                      (-n) 256
...
```

File: components/util/workqueue.rs [hunk 1]
```diff
@@ -76,14 +76,26 @@ struct WorkerThread<QueueData: 'static, WorkData: 'static> {
...
 unsafe impl<QueueData: 'static, WorkData: 'static> Send for WorkerThread<QueueData, WorkData> {}
 
-static SPIN_COUNT: u32 = 128;
-static SPINS_UNTIL_BACKOFF: u32 = 100;
-static BACKOFF_INCREMENT_IN_US: u32 = 5;
+const SPINS_UNTIL_BACKOFF: u32 = 128;
+const BACKOFF_INCREMENT_IN_US: u32 = 5;
+const BACKOFFS_UNTIL_CONTROL_CHECK: u32 = 6;
+
+fn next_power_of_two(mut v: u32) -> u32 {
```
Comment (SimonSapin) [inline components/util/workqueue.rs]: Is this different from the `u32::next_power_of_two` method?

**Pull request merged**